File: packages/java-sdk/src/main/java/com/turbodocx/models/DocumentStatusResponse.java
package com.turbodocx.models;

import com.google.gson.annotations.SerializedName;

/**
 * Response from getting document status
 */
public class DocumentStatusResponse {
    @SerializedName("status")
    private String status;

    @SerializedName("expiresAt")
    private String expiresAt;

    public String getStatus() {
        return status;
    }

    /**
     * ISO timestamp when the signing window closes, if the document has a deadline.
     * Null when expiration is off (the default), which means the document never expires.
     */
    public String getExpiresAt() {
        return expiresAt;
    }
}

File: packages/php-sdk/src/Types/Responses/DocumentStatusResponse.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Responses;

/**
 * Response from getStatus
 *
 * Carries the document status plus, when expiration is enabled, the signing-window
 * deadline (expiresAt). Matches the status/expiresAt shape returned by the JS/Python/Go/Java SDKs.
 */
final class DocumentStatusResponse
{
    /**
     * @param string $status Document status (e.g., 'draft', 'under_review', 'completed', 'voided', 'expired')
     * @param string|null $expiresAt ISO timestamp when the signing window closes, or null when expiration is off
     */
    public function __construct(
        public string $status,
        public ?string $expiresAt = null,
    ) {}

    /**
     * Create from array
     *
     * @param array<string, mixed> $data
     * @return self
     */
    public static function fromArray(array $data): self
    {
        return new self(
            status: $data['status'] ?? '',
            expiresAt: $data['expiresAt'] ?? null,
        );
    }
}
